### PR TITLE
Configure functional testing

### PR DESCRIPTION
**Pull request by ashalak**
According with task [#4](https://github.com/rednavis/vaadin-showcase/issues/4).

### CODE REVIEW

**Comment by AlexOreshkevich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- src/test/java/com/rednavis/vaadin/showcase/backend/BackendServiceTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/rednavis/vaadin-showcase/pullRequest?prid=4808499&bid=15996202)

File: src/test/java/com/rednavis/vaadin/showcase/backend/BackendServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package com.rednavis.vaadin.showcase.backend;
+
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (AlexOreshkevich) [inline src/test/java/com/rednavis/vaadin/showcase/backend/BackendServiceTest.java]: Codacy Issue found: [Avoid unused imports such as 'org.junit.jupiter.api.Assertions'](https://app.codacy.com/gh/rednavis/vaadin-showcase/pullRequest?prid=4808499)

File: src/test/java/com/rednavis/vaadin/showcase/backend/BackendServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  
+  private BackendService service;
+  
+  @BeforeAll
+  public void init() {
+    service = new BackendService();
+  }
+
+  @Test
+  public void should_test_get_employees() {
```
Comment (AlexOreshkevich) [inline src/test/java/com/rednavis/vaadin/showcase/backend/BackendServiceTest.java]: Codacy Issue found: [The JUnit 4 test method name 'should_test_get_employees' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/gh/rednavis/vaadin-showcase/pullRequest?prid=4808499)

File: src/test/java/com/rednavis/vaadin/showcase/backend/BackendServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+
+@TestInstance(Lifecycle.PER_CLASS)
+public class BackendServiceTest {
+  
+  private BackendService service;
+  
+  @BeforeAll
+  public void init() {
+    service = new BackendService();
```
Comment (alabotski) [inline src/test/java/com/rednavis/vaadin/showcase/backend/BackendServiceTest.java]: May be you need use DI?

**Comment by AlexOreshkevich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/com/rednavis/vaadin/showcase/backend/EmployeeTest.java  1
- src/test/java/com/rednavis/vaadin/showcase/backend/BackendServiceTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/rednavis/vaadin-showcase/pullRequest?prid=4808499&bid=15996202)

**Comment by AlexOreshkevich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/com/rednavis/vaadin/showcase/backend/EmployeeTest.java  1
- src/test/java/com/rednavis/vaadin/showcase/backend/BackendServiceTest.java  1
- src/main/java/com/rednavis/vaadin/showcase/views/MainView.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/rednavis/vaadin-showcase/pullRequest?prid=4808499&bid=15996202)

File: src/test/java/com/rednavis/vaadin/showcase/backend/BackendServiceTest.java (hunk 3 above)
Comment (ashalak) [reply src/test/java/com/rednavis/vaadin/showcase/backend/BackendServiceTest.java]: done

**Comment by AlexOreshkevich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/com/rednavis/vaadin/showcase/backend/BackendServiceTest.java  1
- src/test/java/com/rednavis/vaadin/showcase/backend/EmployeeTest.java  1
- src/main/java/com/rednavis/vaadin/showcase/views/MainView.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/rednavis/vaadin-showcase/pullRequest?prid=4808499&bid=15996202)

**Comment by AlexOreshkevich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/com/rednavis/vaadin/showcase/backend/EmployeeTest.java  1
- src/test/java/com/rednavis/vaadin/showcase/backend/BackendServiceTest.java  1
- src/main/java/com/rednavis/vaadin/showcase/views/MainView.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/rednavis/vaadin-showcase/pullRequest?prid=4808499&bid=15996202)

**Comment by AlexOreshkevich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/com/rednavis/vaadin/showcase/backend/EmployeeTest.java  1
- src/main/java/com/rednavis/vaadin/showcase/views/MainView.java  1
- src/test/java/com/rednavis/vaadin/showcase/backend/BackendServiceTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/rednavis/vaadin-showcase/pullRequest?prid=4808499&bid=15996202)

**Comment by AlexOreshkevich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/com/rednavis/vaadin/showcase/backend/BackendServiceTest.java  1
- src/test/java/com/rednavis/vaadin/showcase/backend/EmployeeTest.java  1
- src/main/java/com/rednavis/vaadin/showcase/views/MainView.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/rednavis/vaadin-showcase/pullRequest?prid=4808499&bid=15996202)

**Comment by AlexOreshkevich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/com/rednavis/vaadin/showcase/backend/EmployeeTest.java  1
- src/main/java/com/rednavis/vaadin/showcase/views/MainView.java  1
- src/test/java/com/rednavis/vaadin/showcase/backend/BackendServiceTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/rednavis/vaadin-showcase/pullRequest?prid=4808499&bid=15996202)

**Comment by AlexOreshkevich** [comment]
Please provide an example of class for all tests with Vaadin UI components `com.vaadin.flow.component.Component` and `com.vaadin.flow.component.polymertemplate.PolymerTemplate` involved, like functional test for Vaadin UI client code.

**Comment by AlexOreshkevich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/com/rednavis/vaadin/showcase/backend/BackendServiceTest.java  1
- src/main/java/com/rednavis/vaadin/showcase/views/MainView.java  1
- src/test/java/com/rednavis/vaadin/showcase/backend/EmployeeTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/rednavis/vaadin-showcase/pullRequest?prid=4808499&bid=15996202)

**Pull request merged**